### PR TITLE
scale gradient accumulation steps with train batch size to keep effective batch size about the same

### PR DESCRIPTION
**Pull request by winglian**
# What does this PR do?
goes with https://github.com/huggingface/accelerate/pull/3071

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/trainer.py [hunk 1]
```diff
@@ -1972,8 +1972,18 @@ def train(
...
+        def reduce_batch_size_fn_wrapper(train_batch_size, args):
+            effective_batch_size = args.gradient_accumulation_steps * train_batch_size
+            def reduce_batch_size_fn():
+                nonlocal train_batch_size
+                train_batch_size = train_batch_size // 2
+                args.gradient_accumulation_steps = effective_batch_size // train_batch_size
+                return train_batch_size
+            return reduce_batch_size_fn
+
         inner_training_loop = find_executable_batch_size(
```
Comment (muellerzr) [inline src/transformers/trainer.py:1985]: The `trainer_utils`'s `find_executable_batch_size` will then also need changing.

Also need to be careful about versioning inside that func

**Review by muellerzr — COMMENTED** [review]
Nice start!

File: src/transformers/trainer.py (hunk 1 above)
Comment (muellerzr) [reply src/transformers/trainer.py:1985]: It should be set as `>=0.34.0`

File: src/transformers/trainer.py [hunk 2]
```diff
@@ -1972,8 +1972,18 @@ def train(
...
 
+
+        def reduce_batch_size_fn_wrapper(train_batch_size, args):
+            effective_batch_size = args.gradient_accumulation_steps * train_batch_size
+            def reduce_batch_size_fn():
+                nonlocal train_batch_size
+                train_batch_size = train_batch_size // 2
+                args.gradient_accumulation_steps = effective_batch_size // train_batch_size
+                return train_batch_size
+            return reduce_batch_size_fn
```
[the remark this answers is not in the document]
Comment (winglian) [reply src/transformers/trainer.py:1983]: sorry, I forgot to commit a file. should be there now

File: src/transformers/trainer.py (hunk 1 above)
Comment (winglian) [reply src/transformers/trainer.py:1985]: sorry, I forgot to commit a file. should be there now
